dtls_time.h

```diff
@@ -73,6 +75,11 @@ typedef clock_time_t dtls_tick_t;
 void dtls_clock_init(void);
 void dtls_ticks(dtls_tick_t *t);
 
+/* see https://godbolt.org/z/YchexKaeT */
+#define DTLS_OFFSET_TIME (((clock_time_t)~0) >> 1)
+/** Checks if A is before (or equal) B. Considers 32 bit time overflow */
+#define DTLS_IS_BEFORE_TIME(A, B) ((DTLS_OFFSET_TIME + (B)-(A)) >= DTLS_OFFSET_TIME)
```

Comment:
```suggestion
#define DTLS_IS_BEFORE_TIME(A, B) ((clock_time_t)(DTLS_OFFSET_TIME + (B)-(A)) >= DTLS_OFFSET_TIME)
```
that cast is needed (check with godbolt)

Reply:
Thanks for pointing to that.